### PR TITLE
Feature/pullrequests

### PR DESCRIPTION
**Pull request by siredmar**
Added PR pipeline that gets triggered by Github Webhook. 
Also checks every 30 minutes (e.g. for changes in forks)

The pipeline builds `kyt-cli` and `kyt-api-server` as well as runs `go test` for both projects.

See https://ci4rail.atlassian.net/wiki/spaces/SOF/pages/469991460/Triggering+concourse+resource+check+with+Github+webhook for information on how to create the github webhook.

**Merge this after #9 has been merged.**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,20 +73,58 @@ $ cd kyt-cli
...
+
+Copy `ci/credentials.template.yaml` to `ci/credentials.yaml` and enter the credentials needed. The `github_access_token` needs `write:packages` rights on Github.
+Apply the CI/CD pipeline to Concourse CI using
+```bash
+$ fly -t prod set-pipeline -p kyt-services -c pipeline.yaml -l ci/config.yaml  -l ci/credentials.yaml
+```
+
+## pipeline-pullrequests.yaml
+
+The `pipeline-pullrequests.yaml` is the CI/CD pipeline that checks Github for commits within Pull Requests and performs a clean build of kyt-cli `kyt` and `kyt-api-server` go binaries. It also runs `go test` for both.
```
Comment (KlausPopp) [inline README.md]: Better: "The `pipeline-pullrequests.yaml` defines a pipeline that runs basic quality checks for each new pull request. For this, consourse checks Github for new pull requests. If a new pull request is found, it downloads the repo, performs a clean build of kyt-cli `kyt` and `kyt-api-server` go binaries. It also runs `go test` for both.

File: README.md [hunk 2]
```diff
@@ -1,14 +1,12 @@
 # kyt-cli
-This repository contains kyt-cli and kyt-api-server sources, build environment and ci/cd pipeline (ci/cd currently missing).
+[![CI](https://concourse.ci4rail.com/api/v1/teams/main/pipelines/kyt-services/jobs/build-kyt-services/badge)](https://concourse.ci4rail.com/teams/main/pipelines/kyt-services) [![Go Report Card](https://goreportcard.com/badge/github.com/ci4rail/kyt-cli)](https://goreportcard.com/report/github.com/ci4rail/kyt-cli)
```
Comment (KlausPopp) [inline README.md]: Cool!

File: README.md [hunk 3]
```diff
@@ -1,14 +1,12 @@
 # kyt-cli
```
Comment (KlausPopp) [inline README.md]: Move sentence "Get the iot hub connection string from the Azure Portal. Select the iot hub, then "shared access policies"" into server docker section,

File: pipeline-pullrequests.yaml [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+              cd ${ROOT}/pull-request/kyt-cli
+              make
+              go test
+      on_failure:
+        put: pull-request
+        params:
+          path: pull-request
+          status: failure
+
+    - task: build-kyt-api-server
```
Comment (KlausPopp) [inline pipeline-pullrequests.yaml:117]: Add build of test server

**Review by KlausPopp — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (siredmar) [reply README.md]: It does not only look for new PRs. It also looks for new PRs, but it also looks for commits in previously active PRs. 
Will rephrase it to:
The `pipeline-pullrequests.yaml` defines a pipeline that runs basic quality checks on pull requests. For this, consourse checks Github for new or changed pull requests If a change is found, it downloads the branch and performs a clean build of kyt-cli `kyt` and `kyt-api-server` go binaries. It also runs go test for both.

File: README.md (hunk 2 above)
Comment (siredmar) [reply README.md]: Indeed.

File: README.md (hunk 3 above)
Comment (siredmar) [reply README.md]: Moved and rephrased: 
Get the IoT Hub connection string from the Azure Portal. Select the IoT Hub, then "shared access policies". Copy from `iothubowner` the connection string `Connection string—primary key`.

File: pipeline-pullrequests.yaml (hunk 4 above)
Comment (siredmar) [reply pipeline-pullrequests.yaml:117]: Test server currently doesn't build. Needs to be done in another PR.

Comment (KlausPopp) [reply pipeline-pullrequests.yaml:117]: Ok

**Review by KlausPopp — APPROVED** [review]
